SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -6761,7 +6874,140 @@ void CEulerSolver::SetPrimitive_Limiter(CGeometry *geometry, CConfig *config) {
     delete [] LocalMaxPrimitive; delete [] GlobalMaxPrimitive;
     
   }
-  
+
+    /*--- Venkatakrishnan Modified limiter ---*/
+    /*---  Simple and Parameter-Free Second Slope Limiter for Unstructured Grid Aerodynamic Simulations. AIAA-J 2012 ---*/
+
+    if (config->GetKind_SlopeLimit_Flow() == VENKATAKRISHNAN_2NDLIM) {
+    
+    /*-- Local Variables ---*/
+    unsigned short nNeigh, iNeigh;
+    unsigned long NumNeigh;
+    su2double Local_Mach_i, Local_Mach_j,  *Local_Primitive, velocity2;
+    su2double f_Mach_i, f_Mach_j;
+    
+    for (iEdge = 0; iEdge < geometry->GetnEdge(); iEdge++) {
+        
+        iPoint     = geometry->edge[iEdge]->GetNode(0);
+        jPoint     = geometry->edge[iEdge]->GetNode(1);
+        Gradient_i = node[iPoint]->GetGradient_Primitive();
+        Gradient_j = node[jPoint]->GetGradient_Primitive();
+        Coord_i    = geometry->node[iPoint]->GetCoord();
+        Coord_j    = geometry->node[jPoint]->GetCoord();
+        
+        
+        AD::StartPreacc();
+        AD::SetPreaccIn(Gradient_i, nPrimVarGrad, nDim);
+        AD::SetPreaccIn(Gradient_j, nPrimVarGrad, nDim);
+        AD::SetPreaccIn(Coord_i, nDim); AD::SetPreaccIn(Coord_j, nDim);
+        
+        /*--- Simple and Parameter-Free Second Slope Limiter for Unstructured Grid Aerodynamic Simulations---*/
```

Comment:
Having a parameter-free option is interesting.. folks still have trouble tuning limiters and avoiding limit cycles/stalled convergence. Have you had a positive experience with this limiter in terms of convergence?

Reply:
There is a trick with this parameter-free second limiter. If you have a high quality 3D grid, this limiter will greatly improve convergence. However for complex geometries (most of the time in aerospace industry), this second limiter will diverge the solution because it will turn off the primary limiter in regions of poor quality cells when M<1.0.
There is a measure of the cell quality (Angle, Skewness, etc) in SU2? If we have,  I would like to use  here in order to avoid turning off the limiter in this regions.

Reply:
I see some problems in the current way it is implemented. First of all doing a search involving neighbors shouldn't be done inside an edge loop as you compute the same values twice (just an efficiency thing). Furthermore, in order for the computation to give the same result using different number of processors (which we should aim for) a communication of the variable Mach_Local is necessary.

Unless you think that this limiter greatly improves convergence, I would suggest to wait with this limiter implementation until we have figured out a way of generalizing this.